pkg/initializer/initializer.go

```diff
@@ -104,32 +99,36 @@ func (e *EtcdInitializer) restoreCorruptData() error {
 
 	e.Config.RestoreOptions.BaseSnapshot = *baseSnap
 	e.Config.RestoreOptions.DeltaSnapList = deltaSnapList
+	tempRestoreOptions := *e.Config.RestoreOptions
+	tempRestoreOptions.RestoreDataDir = fmt.Sprintf("%s.%s", tempRestoreOptions.RestoreDataDir, "part")
 
-	rs := restorer.NewRestorer(store, logger)
+	logger.Infof("Removing data directory(%s) for snapshot restoration.", tempRestoreOptions.RestoreDataDir)
+	if err := os.RemoveAll(filepath.Join(tempRestoreOptions.RestoreDataDir)); err != nil {
+		return fmt.Errorf("failed to delete member directory: %v", err)
```

Comment:
Change the error message: `fmt.Errorf("failed to remove temporary data directory %s with err: %v",dataDir, err)`